src/symmetric_crypto/aes_256_gcm_pure/mod.rs

```diff
@@ -47,11 +47,10 @@ impl Dem<KEY_LENGTH> for Aes256GcmCrypto {
         additional_data: Option<&[u8]>,
     ) -> Result<Vec<u8>, CryptoCoreError> {
         if plaintext.len() as u64 > MAX_PLAINTEXT_LENGTH {
-            return Err(CryptoCoreError::InvalidSize(format!(
-                "Plaintext is too large ({} bytes), max size: {} ",
-                plaintext.len(),
-                MAX_PLAINTEXT_LENGTH
-            )));
+            return Err(CryptoCoreError::PlaintextTooBigError {
+                plaintext_len: plaintext.len(),
+                max: MAX_PLAINTEXT_LENGTH,
+            });
```

Comment:
I would not create a dedicated error, rather returning an `EncryptionError`.

Reply:
Can this leak information? Because I think it's a really useful error for the user of the lib to known that what it's sent is too big…

Reply:
I think we're leaking this information with the timing anyway… (since it's a early return)

Reply:
Indeed, a timing check could lead to this information. You can keep it like this if you prefer. Same for my other comments.